src/interp/interp.cc

```diff
@@ -2390,6 +2443,87 @@ RunResult Thread::DoAtomicRmwCmpxchg(Instr instr, Trap::Ptr* out_trap) {
   return RunResult::Ok;
 }
 
+RunResult Thread::DoThrow(Ref exn_ref) {
+  Istream::Offset target_offset = Istream::kInvalidOffset;
+  Exception::Ptr exn{store_, exn_ref};
+  Tag::Ptr exn_tag{store_, exn->tag()};
+  bool popped_frame = false;
+  bool had_catch_all = false;
+
+  // DoThrow is responsible for unwinding the stack at the point at which an
+  // exception is thrown, and also branching to the appropriate catch within
+  // the target try-catch. In a compiler, the tag dispatch might be done in
+  // generated code in a landing pad, but this is easier for the interpreter.
+  while (!frames_.empty()) {
+    const Frame& frame = frames_.back();
+    DefinedFunc::Ptr func{store_, frame.func};
+    u32 pc = frame.offset;
+    auto handlers = func->desc().handlers;
+
+    // We iterate in reverse order, in order to traverse handlers from most
+    // specific (pushed last) to least specific within a nested stack of
+    // try-catch blocks.
+    auto iter = handlers.rbegin();
+    while (iter != handlers.rend()) {
+      const HandlerDesc& handler = *iter;
+      if (pc >= handler.start_offset && pc < handler.end_offset) {
+        // For a try-delegate, skip part of the traversal by directly going
+        // up to an outer handler specified by the delegate depth.
+        if (handler.kind == HandlerKind::Delegate) {
+          iter = handlers.rend() - handler.delegate_offset - 1;
```

Comment:
In this case, since we're iterating in reverse and want to potentially continue iterating after the delegating (the delegated handler may not handle the tag, etc), we want `iter` to be a reverse iterator. Since the handler index is from the start, we subtract from `rend()` which is equivalent to the start in the other direction (but off by one, because `rend()` is off the end instead of the first element).

I can add a comment to make this clearer.